test/e2e/k8s_conformance_test.go

```diff
@@ -40,17 +43,16 @@ var _ = Describe("When testing K8S conformance [Conformance] [K8s-Install]", fun
 })
 
 var _ = Describe("When testing K8S conformance with K8S latest ci [Conformance] [K8s-Install-ci-latest]", func() {
+	// Note: This installs a cluster based on KUBERNETES_VERSION_LATEST_CI and runs conformance tests.
+	// Note: We are resolving KUBERNETES_VERSION_LATEST_CI and then setting the resolved version as
+	// KUBERNETES_VERSION env var. This only works without side effects on other tests because we are
+	// running this test in its separate job.
 	K8SConformanceSpec(ctx, func() K8SConformanceSpecInput {
-		kubernetesVersion, err := kubernetesversions.ResolveVersion(ctx, e2eConfig.Variables["KUBERNETES_VERSION_LATEST_CI"])
+		kubernetesVersion, err := kubernetesversions.ResolveVersion(ctx, e2eConfig.GetVariable("KUBERNETES_VERSION_LATEST_CI"))
 		Expect(err).NotTo(HaveOccurred())
-
-		// Kubernetes version has to be set as KUBERNETES_VERSION because the conformance test
-		// expects it there.
-		testSpecificE2EConfig := e2eConfig.DeepCopy()
-		e2eConfig.Variables["KUBERNETES_VERSION"] = kubernetesVersion
-
+		Expect(os.Setenv("KUBERNETES_VERSION", kubernetesVersion)).To(Succeed())
 		return K8SConformanceSpecInput{
```

Comment:
The actual mistake that lead to us using the wrong version was setting the variable in e2eConfig instead of in testSpecificE2EConfig :joy: